### PR TITLE
Allow action to set semver based on tag, without hosting project complaining 

### PR DESCRIPTION
**Pull request by bjornrud**
Connects https://github.com/NFIBrokerage/technical_ideas/issues/116

### CODE REVIEW

File: mix.exs [hunk 1]
```diff
@@ -1,9 +1,12 @@
 defmodule Hummingbird.MixProject do
   use Mix.Project
 
-  {git_tag, _return_code} = System.cmd("git", ["describe", "--abbrev=0", "--tags"])
+  @version_file Path.join(__DIR__, ".library_version")
 
-  @version (case Regex.run(~r/^v([\d\.]+)/, git_tag, capture: :all_but_first) do
+  # a special module attribute that recompiles if targetted file has changed
```
Comment (the-mikedavis) [inline mix.exs:6]:
```suggestion
  # a special module attribute that recompiles if targeted file has changed
```

I think?

**Comment by bjornrud** [comment]
@pmonson711 we sharing, just 'cause we thought you might be interested for your own libs how we're versioning the hex package, relatively simply.

**Comment by pmonson711** [comment]
@bjornrud This is interesting, but I'm curious about moving it away from the `mix.exs` solution? I tend to like things I can run/test locally over the CI only solutions

**Comment by bjornrud** [comment]
@pmonson711 I'm sure I'm missing something here.  What type of tests are you looking to get locally?  This does indeed work that way.  

If you change the library version file locally it does, report that version on compilation.

Like a test that actually manipulates the version file and checks that version is indeed as expected?

**Comment by pmonson711** [comment]
@bjornrud I think I used the wrong term with "test". 

It's not important here but I just trying to state that I could run the `git describe...` locally but the 'Write tagged version to .library_version to ensure published' depends on Github infra. I almost always push the functionality into the local build system. Given the way git manages tags, I think this actually makes sense after thinking more. 
:+1:

**Pull request merged**